File: yt/yt/flow/examples/java/retryable_async_request/retryable_async_request/src/main/java/tech/ytsaurus/flow/examples/retryableasyncrequest/model/RequestState.java
package tech.ytsaurus.flow.examples.retryableasyncrequest.model;

import javax.persistence.Column;
import javax.persistence.Entity;

/**
 * Internal state for a single request tracked by the processor.
 * Serialized as YSON for internal state storage.
 */
// [BEGIN request_state]
@Entity
public class RequestState {
    @Column(name = "request_id")
    private long requestId;

    private long key;

    private String request;

    @Column(name = "failed_attempts")
    private int failedAttempts;

    public RequestState() {
    }

    public RequestState(long requestId, long key, String request, int failedAttempts) {
        this.requestId = requestId;
        this.key = key;
        this.request = request;
        this.failedAttempts = failedAttempts;
    }

    public long getRequestId() {
        return requestId;
    }

    public void setRequestId(long requestId) {
        this.requestId = requestId;
    }

    public long getKey() {
        return key;
    }

    public void setKey(long key) {
        this.key = key;
    }

    public String getRequest() {
        return request;
    }

    public void setRequest(String request) {
        this.request = request;
    }

    public int getFailedAttempts() {
        return failedAttempts;
    }

    public void setFailedAttempts(int failedAttempts) {
        this.failedAttempts = failedAttempts;
    }

    @Override
    public String toString() {
        return "RequestState{" +
                "requestId=" + requestId +
                ", key=" + key +
                ", request='" + request + '\'' +
                ", failedAttempts=" + failedAttempts +
                '}';
    }
}
// [END request_state]

File: yt/yt/flow/examples/kotlin/retryable_async_request/retryable_async_request/src/main/kotlin/tech/ytsaurus/flow/examples/retryableasyncrequest/model/RequestState.kt
package tech.ytsaurus.flow.examples.retryableasyncrequest.model

import javax.persistence.Column
import javax.persistence.Entity

// [BEGIN request_state]
@Entity
class RequestState {
    @Column(name = "request_id")
    var requestId: Long = 0

    var key: Long = 0

    var request: String? = null

    @Column(name = "failed_attempts")
    var failedAttempts: Int = 0

    constructor()

    constructor(requestId: Long, key: Long, request: String?, failedAttempts: Int) {
        this.requestId = requestId
        this.key = key
        this.request = request
        this.failedAttempts = failedAttempts
    }

    override fun toString(): String {
        return "RequestState{" +
            "requestId=" + requestId +
            ", key=" + key +
            ", request='" + request + '\'' +
            ", failedAttempts=" + failedAttempts +
            '}'
    }
}
// [END request_state]
